xbatcher/tests/test_generators.py

```diff
@@ -160,3 +167,22 @@ def test_batch_3d_2d_input_concat(sample_ds_3d, bsize):
             * (sample_ds_3d.dims['y'] // bsize)
             * sample_ds_3d.dims['time']
         )
+
+
+def test_preload_batch_false(sample_ds_1d):
+    sample_ds_1d_dask = sample_ds_1d.chunk({'x': 2})
+    bg = BatchGenerator(
+        sample_ds_1d_dask, input_dims={'x': 2}, preload_batch=False
+    )
+    assert bg.preload_batch is False
+    for ds_batch in bg:
+        assert isinstance(ds_batch, xr.Dataset)
+        assert ds_batch.chunks
+
+
+def test_preload_batch_true(sample_ds_1d):
+    bg = BatchGenerator(sample_ds_1d, input_dims={'x': 2}, preload_batch=True)
```

Comment:
do we need to add `sample_ds_1d_dask = sample_ds_1d.chunk({'x': 2})` before this line as well to make sure this test will fail with with `preload_batch=False`?